### PR TITLE
Remove deleted views

### PR DESCRIPTION
**Pull request by manutarus**
remove deleted views from form json

### CODE REVIEW

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java [hunk 1]
```diff
@@ -72,6 +74,7 @@ public AttachRepeatingGroupTask(final ViewParent parent, int numRepeatingGroups,
     @Override
     protected void onPreExecute() {
         showProgressDialog(R.string.please_wait_title, R.string.creating_repeating_group_message, widgetArgs.getFormFragment().getContext());
+        repeatingGroups = new ArrayList<>();
```
Comment (bennsimon) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java]: What is this arrayList for?

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java [hunk 2]
```diff
@@ -100,14 +104,27 @@ protected void onPostExecute(List<View> result) {
...
+                ArrayList<String> removeThisFields = new ArrayList<>();
                 int len = fields.length();
                 for (int i = len - 1; i >= 0; i--) {
                     String[] key = ((String) fields.getJSONObject(i).get(KEY)).split("_");
                     if (keysToRemove.contains(key[key.length - 1])) {
+                        removeThisFields.add((String) fields.getJSONObject(i).get(KEY));
                         fields.remove(i);
                     }
                 }
+//                remove deleted views from form json
```
Comment (bennsimon) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java]: I think the comment here can be updated

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java [hunk 3]
```diff
@@ -100,14 +104,27 @@ protected void onPostExecute(List<View> result) {
...
+                Collection<View> viewCollection =  widgetArgs.getFormFragment().getJsonApi().getFormDataViews();
+                Iterator<View> viewIterator = viewCollection.iterator();
+                while (viewIterator.hasNext()){
+                    View view = viewIterator.next();
+                    String key = (String) view.getTag(R.id.key);
+                    if(removeThisFields.contains(key)){
```
Comment (bennsimon) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java]: This method seems reusable you can add it in utils

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java [hunk 4]
```diff
@@ -100,14 +104,27 @@ protected void onPostExecute(List<View> result) {
...
                 int len = fields.length();
                 for (int i = len - 1; i >= 0; i--) {
                     String[] key = ((String) fields.getJSONObject(i).get(KEY)).split("_");
                     if (keysToRemove.contains(key[key.length - 1])) {
+                        removeThisFields.add((String) fields.getJSONObject(i).get(KEY));
                         fields.remove(i);
                     }
                 }
+//                remove deleted views from form json
+                Collection<View> viewCollection =  widgetArgs.getFormFragment().getJsonApi().getFormDataViews();
```
Comment (bennsimon) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java]: I think you should check for null in this chained method calls `widgetArgs.getFormFragment().getJsonApi().getFormDataViews()`

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java (hunk 1 above)
Comment (manutarus) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java]: Thanks for catching that, I have removed

**Pull request merged**